src/tracker-component.js

```diff
@@ -0,0 +1,40 @@
+/* @flow */
+
+import { getClientID } from '@paypal/sdk-client/src';
+
+const user = { id: undefined, name: undefined };
+
+const track = (trackingType, trackingData) => {
+    const encodeData = data => encodeURIComponent(btoa(JSON.stringify(data)));
+
+    return new Promise((resolve, reject) => {
+        const img = document.createElement('img');
+        img.style.display = 'none';
+        img.src = `https://api.keen.io/3.0/projects/5b903f6ec9e77c00013bc6a7/events/${ trackingType }?api_key=700B56FBE7A2A6BD845B82F9014ED6628943091AD5B0A5751C3027CFE8C5555448C6E11302BD769FCC5BDD003C3DE8282C4FC8FE279A0AAC866F2C97010468197B98779B872EFD7EE980D2986503B843DA3797C750DAA00017DC8186078EADA6&data=${ encodeData(
+            {
+                ...trackingData,
+                userId:   user.id,
+                userName: user.name,
+                clientId: getClientID()
+            }
+        ) }`;
+        img.addEventListener('load', resolve);
+        img.addEventListener('error', reject);
+        document.body.appendChild(img);
+    });
+};
+
+const trackCartEvent = (cartEventType, trackingData) =>
+    track('cartEvent', { ...trackingData, cartEventType });
+
+export const Tracker = {
```

Comment:
Would it make sense to do something like:

```
let cartTracker = paypal.CartTracker({ userID, userName })
cartTracker.addToCart();
```

That would avoid the need to use a global `user`, and also makes the module less of a singleton / potentially extendable in future if there are any partners or marketplaces who have multiple carts, or anything like that.

On that node -- seems like it would be a good idea to accept a `cartID` in that constructor here so we can disambiguate multiple carts in future, especially for partner cases.

Reply:
:+1: I second the above style: `let cartTracker = paypal.CartTracker({ userID, userName })`

Reply:
The idea is to track all events on a merchant's page, not just cart events. It really is the FPTI client for merchants to implement. I am working on the docs here https://github.paypal.com/arnaidu/tracker/blob/master/PayPalShopping.md

I feel we should not mandate 3 different objects ... PageTracker, CartTracker, PurchaseTracker in order to track events. Instead we were thinking one instance Tracker that one uses like this. Tracker.addToCart, Tracker.purchase, Tracker.pageView.

```

Reply:
All events will have to carry the clientId and merchantId if available. This will solve for the partner and market places no?

Reply:
@arvindrajnaidu I agree with you. I was second'ing the "let's not set a global user" thing.

Reply:
That's fair. If we're going to use this for multiple use-cases let's go with `paypal.Tracker`. But let's make it a constuctor (`let tracker = paypal.Tracker({ userID, userName })`)